base/tests/integrationtests/com/google/idea/blaze/base/run/producer/BlazeBuildFileRunLineMarkerProviderTest.java

```diff
@@ -0,0 +1,42 @@
+package com.google.idea.blaze.base.run.producer;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.base.Objects;
+import com.google.idea.blaze.base.lang.buildfile.psi.util.PsiUtils;
+import com.google.idea.blaze.base.model.primitives.WorkspacePath;
+import com.google.idea.blaze.base.run.producers.BlazeBuildFileRunLineMarkerProvider;
+import com.intellij.execution.lineMarker.RunLineMarkerContributor.Info;
+import com.intellij.icons.AllIcons;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.impl.source.tree.LeafPsiElement;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Integration tests for {@link BlazeBuildFileRunLineMarkerProvider} */
+@RunWith(JUnit4.class)
+public class BlazeBuildFileRunLineMarkerProviderTest extends BlazeRunConfigurationProducerTestCase {
+
+  @Test
+  public void testFunCallInfo() {
+    BlazeBuildFileRunLineMarkerProvider markerContributor =
+        new BlazeBuildFileRunLineMarkerProvider();
+
+    PsiFile buildFile =
+        createAndIndexFile(WorkspacePath.createIfValid("BUILD"), "java_library(name = 'bar')");
+    List<LeafPsiElement> elements =
+        PsiUtils.findAllChildrenOfClassRecursive(buildFile, LeafPsiElement.class);
+    LeafPsiElement funCallIdentifier =
+        elements.stream()
+            .filter(e -> Objects.equal(e.getText(), "java_library"))
```

Comment:
We should only run for binaries and tests.

Reply:
Do you mean that we should only add a gutter icons for binaries and tests e.g. not java_libraries? I find it helpful to `bazel build` java_libraries in order to check for compiler errors. 

I borrowed some logic from `BlazeBuildFileRunConfigurationProducer` in order to prevent placing icons on non-runnable functions like `load()` and `package()`.